src/core/lookup.c

```diff
@@ -360,8 +360,9 @@ QuicLookupFindConnectionByLocalCidInternal(
         // partitioned hash table array, and look up the connection in that
         // hash table.
         //
-        QUIC_STATIC_ASSERT(MSQUIC_CID_PID_LENGTH == 1, "The code below assumes 1 byte");
-        uint32_t PartitionIndex = CID[MsQuicLib.CidServerIdLength];
+        QUIC_STATIC_ASSERT(MSQUIC_CID_PID_LENGTH == 2, "The code below assumes 2 bytes");
+        uint32_t PartitionIndex;
```

Comment:
I think we can just change this to a `uint16_t`.
```suggestion
        uint16_t PartitionIndex;
```